### PR TITLE
chore(network_info_plus): Update Kotlin, Gradle and add Dependabot config

### PR DESCRIPTION
**Pull request by vbuberen**
## Description

Same as #1261 - updating Kotlin, Gradle and adding a Dependabot config for Android dependencies.

## Related Issues

#1248 
#1222 

## Checklist

- [x] I read the Contributor Guide and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -43,8 +43,8 @@ updates:
       prefix: "chore"
       include: "scope"
 
- # Android Intent Plus dependencies updates config
-      
+      # Android Intent Plus dependencies updates config
```
Comment (nohli) [inline .github/dependabot.yml:46]: Why all the different indentations?

**Review by nohli — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (vbuberen) [reply .github/dependabot.yml]: Because of autoformat

Comment (vbuberen) [reply .github/dependabot.yml]: Is it a blocker? And it is fixed already, by the way.

**Review by nohli — APPROVED** [review]

File: .github/dependabot.yml [hunk 2]
```diff
@@ -43,7 +43,7 @@ updates:
       prefix: "chore"
       include: "scope"
 
- # Android Intent Plus dependencies updates config
+# Android Intent Plus dependencies updates config
```
Comment (nohli) [inline .github/dependabot.yml:46]: This change in whitespace can be discarded to match the other indentations.

**Review by nohli — APPROVED** [review]

**Review by miquelbeltran — APPROVED** [review]

**Pull request merged**